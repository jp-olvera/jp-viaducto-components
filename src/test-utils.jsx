import React from 'react';
import { render } from '@testing-library/react';
import { ConfigProvider } from './providers';

const AllTheProviders = ({ children }) => {
  return <ConfigProvider>{children}</ConfigProvider>;
};

const customRender = (ui, options) =>
  render(ui, { wrapper: AllTheProviders, ...options });

// re-export everything
export * from '@testing-library/react';

// override render method
export { customRender as render };
