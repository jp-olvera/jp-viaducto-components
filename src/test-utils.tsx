import React from 'react';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom';
import { ConfigProvider } from './providers';

const AllTheProviders = ({ children }: any) => (
  <ConfigProvider>{children}</ConfigProvider>
);

jest.mock('./cells/Input/amex.svg', () => null);
jest.mock('./cells/Input/mastercard.svg', () => null);
jest.mock('./cells/Input/visa.svg', () => null);
jest.mock('./cells/Dropdown/sorting.svg', () => null);
jest.mock('./cells/BareButton/close.svg', () => null);

const customRender = (ui, options?) => render(ui, { wrapper: AllTheProviders, ...options });
// re-export everything
export * from '@testing-library/react';

// override render method
export { customRender as render };
