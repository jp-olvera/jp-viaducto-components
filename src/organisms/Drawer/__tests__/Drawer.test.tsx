/* eslint-env jest */

import React from 'react';
import '@testing-library/jest-dom/extend-expect';
import { render, screen, fireEvent } from '../../../test-utils';
import { Drawer } from '..';

describe('<Drawer/>', () => {
  test('should be visible', () => {
    render(
      <Drawer active onClose={() => {}}>
        <h1>Title</h1>
      </Drawer>,
    );
    expect(screen.getByText('Title')).toBeVisible();
  });
  test('should no be visible', () => {
    render(
      <Drawer active={false} onClose={() => {}}>
        <h1>Title</h1>
      </Drawer>,
    );
    expect(screen.queryByText('Title')).toMatchSnapshot();
  });
  test('should call onClose when clicking the overlay', () => {
    jest.useFakeTimers();
    const onClose = jest.fn();
    render(
      <Drawer onClose={onClose} active>
        <h1>Title</h1>
      </Drawer>,
    );
    fireEvent.click(screen.getByTestId('overlay'));
    jest.runOnlyPendingTimers();
    expect(onClose).toBeCalledTimes(1);
    jest.useRealTimers();
  });
});
