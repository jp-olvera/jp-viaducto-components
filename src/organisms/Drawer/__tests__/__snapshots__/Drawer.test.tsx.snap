// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Drawer/> should click drawer 1`] = `
<div
  class="StyledDrawer-sc-etbwfh-0 hcUVdE"
  data-testid="drawer"
  elevation="1"
  tabindex="0"
  width="22.25rem"
>
  <h1>
    Title
  </h1>
</div>
`;

exports[`<Drawer/> should fire keydown on overlay 1`] = `
<div
  data-testid="overlay"
  role="dialog"
  style="background-color: rgba(0, 0, 0, 0.3); left: 0px; height: 100vh; position: fixed; top: 0px; width: 100vw; z-index: 1;"
>
  <div
    class="StyledDrawer-sc-etbwfh-0 blqnxU"
    data-testid="drawer"
    elevation="1"
    tabindex="0"
    width="33.375rem"
  >
    <h1>
      Title
    </h1>
  </div>
</div>
`;

exports[`<Drawer/> should no be visible 1`] = `null`;
